packages/@aws-cdk/aws-iam/lib/principals.ts

```diff
@@ -153,7 +181,11 @@ export class PrincipalWithConditions implements IPrincipal {
   }
 
   public addToPolicy(statement: PolicyStatement): boolean {
-    return this.principal.addToPolicy(statement);
+    return this.addToIdentityPolicy(statement).statementAdded;
+  }
+
+  public addToIdentityPolicy(statement: PolicyStatement): AddToIdentityPolicyResult {
```

Comment:
Should these be called `addToPrincipalPolicy`?

Reply:
Maybe. Right now it's:

* **Principal**: anything you can grant permissions to, that goes into a policy (user, role, but also service principal or account principal)
* **Identity**: an IAM identity you create (user, role, group).

Policies only go on Identities, for example "adding to a service principal's policy" doesn't make any sense.

However, I will grant that this distinction might not be obvious, and it might confuse more than it helps.

Thoughts?

Reply:
I am okay with that distinction although I also think it's not 100% required in this case and `addToPrincipalPolicy` would have been a bit clearer